// react-game-kit/lib/utils/audio-player.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

/* eslint-disable no-console */
var AudioPlayer = function AudioPlayer(url, callback) {
  var _this = this;

  _classCallCheck(this, AudioPlayer);

  this.play = function (options) {
    var volume = options && options.volume;
    var offset = options && options.offset;
    var loop = options && options.loop;

    var source = _this.context.createBufferSource();
    var gainNode = _this.context.createGain();
    gainNode.gain.value = volume || 0.5;

    gainNode.connect(_this.context.destination);
    source.connect(gainNode);

    source.buffer = _this.buffer;
    source.start(offset ? _this.context.currentTime + offset : 0);
    source.loop = loop || false;
    return source.stop.bind(source);
  };

  this.loadBuffer = function () {
    var request = new XMLHttpRequest();
    request.open('GET', _this.url, true);
    request.responseType = 'arraybuffer';

    request.onload = function () {
      _this.context.decodeAudioData(request.response, function (buffer) {
        if (!buffer) {
          console.error('error decoding file data: ' + _this.url);
          return;
        }
        _this.buffer = buffer;
        _this.callback();
      }, function (error) {
        console.error('decodeAudioData error', error);
      });
    };

    request.onerror = function onError() {
      console.error('BufferLoader: XHR error');
    };

    request.send();
  };

  this.url = url || null;
  this.callback = callback || function () {};

  this.buffer = null;

  window.AudioContext = window.AudioContext || window.webkitAudioContext;
  this.context = window.context || new AudioContext();

  this.loadBuffer();
};

exports.default = AudioPlayer;
;

(function () {
  if (typeof __REACT_HOT_LOADER__ === 'undefined') {
    return;
  }

  __REACT_HOT_LOADER__.register(AudioPlayer, 'AudioPlayer', 'src/utils/audio-player.js');
})();

;
