// react-game-kit/lib/utils/game-loop.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

var GameLoop = function () {
  function GameLoop() {
    var _this = this;

    _classCallCheck(this, GameLoop);

    this.loop = function () {
      _this.subscribers.forEach(function (callback) {
        callback.call();
      });

      _this.loopID = window.requestAnimationFrame(_this.loop);
    };

    this.subscribers = [];
    this.loopID = null;
  }

  _createClass(GameLoop, [{
    key: "start",
    value: function start() {
      if (!this.loopID) {
        this.loop();
      }
    }
  }, {
    key: "stop",
    value: function stop() {
      window.cancelAnimationFrame(this.loopID);
    }
  }, {
    key: "subscribe",
    value: function subscribe(callback) {
      return this.subscribers.push(callback);
    }
  }, {
    key: "unsubscribe",
    value: function unsubscribe(id) {
      this.subscribers.splice(id - 1, 1);
    }
  }]);

  return GameLoop;
}();

exports.default = GameLoop;
;

(function () {
  if (typeof __REACT_HOT_LOADER__ === 'undefined') {
    return;
  }

  __REACT_HOT_LOADER__.register(GameLoop, "GameLoop", "src/utils/game-loop.js");
})();

;
