// react-game-kit/lib/utils/key-listener.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

var KeyListener = function KeyListener() {
  var _this = this;

  _classCallCheck(this, KeyListener);

  this.LEFT = 37;
  this.RIGHT = 39;
  this.UP = 38;
  this.DOWN = 40;
  this.SPACE = 32;

  this.down = function (event) {
    if (event.keyCode in _this.keys) {
      event.preventDefault();
      _this.keys[event.keyCode] = true;
    }
  };

  this.up = function (event) {
    if (event.keyCode in _this.keys) {
      event.preventDefault();
      _this.keys[event.keyCode] = false;
    }
  };

  this.isDown = function (keyCode) {
    return _this.keys[keyCode] || false;
  };

  this.subscribe = function (keys) {
    window.addEventListener('keydown', _this.down);
    window.addEventListener('keyup', _this.up);

    keys.forEach(function (key) {
      _this.keys[key] = false;
    });
  };

  this.unsubscribe = function () {
    window.removeEventListener('keydown', _this.down);
    window.removeEventListener('keyup', _this.up);
    _this.keys = {};
  };

  this.keys = {};
};

exports.default = KeyListener;
;

(function () {
  if (typeof __REACT_HOT_LOADER__ === 'undefined') {
    return;
  }

  __REACT_HOT_LOADER__.register(KeyListener, 'KeyListener', 'src/utils/key-listener.js');
})();

;
